client/db/bolt/upgrades.go

```diff
@@ -0,0 +1,122 @@
+// This code is available on the terms of the project LICENSE.md file,
+// also available online at https://blueoakcouncil.org/license/1.0.0.
+
+package bolt
+
+import (
+	"fmt"
+
+	"decred.org/dcrdex/dex/encode"
+	"go.etcd.io/bbolt"
+)
+
+const (
+	initialVersion = 0
+
+	// versionedDBVersion is the second version of the database.
+	// The only change for this version is that the DB now stores
+	// it's version number.
+	versionedDBVersion = 1
```

Comment:
The name really threw me off, but I understand that the variable's name is of the format `descriptionofDBVersion`.  But I don't see any use of this constant outside of the `upgrades` array.  So why not define the upgrades array implicitly and infer the target version:

```go
type upgradefunc func(tx *bbolt.Tx) error

// Each database upgrade function should be keyed by the database
// version it upgrades.
var upgrades = [...]upgradefunc{
	upgradeToV1, // v0 => v1 adds a version key
}

// DBVersion is the latest version of the database that is understood. Databases
// with recorded versions higher than this will fail to open (meaning any
// upgrades prevent reverting to older software).
const DBVersion = uint32(len(upgrades))
```